### PR TITLE
Revamp [Gen 6] Random Battle

### PR DESCRIPTION
**Pull request by livid-washed**
Framework and `random-teams.ts` code by @livid-washed; they are minor modifications from Gen 7 Random Battle.
Sets created by the Random Battles room staff, lead by @ACakeWearingAHat

This is finished and ready for review.

This pull request completely overhauls Gen 6 Random Battles to utilize the more modern and streamlined Gen 9/7 Random Battles setgen system, allowing us to have split sets, frontloaded move generation, and role-based move and item enforcement logic.

### CODE REVIEW

**Review by ACakeWearingAHat — APPROVED** [review]
gave this a look over and couldn't find anything off about how this is compared to 7

File: data/mods/gen6/random-teams.ts [hunk 1]
```diff
@@ -1,520 +1,520 @@
...
-				!counter.get('Flying') && (
-					abilities.has('Gale Wings') ||
-					abilities.has('Serene Grace') ||
-					(types.has('Normal') && movePool.includes('bravebird'))
-				)
+			Fairy: (movePool, moves, abilities, types, counter) => !counter.get('Fairy'),
+			Fighting: (movePool, moves, abilities, types, counter) => !counter.get('Fighting'),
+			Fire: (movePool, moves, abilities, types, counter) => !counter.get('Fire'),
+			Flying: (movePool, moves, abilities, types, counter, species) => (
+				!counter.get('Flying') && ['aerodactylmega', 'charizardmegay', 'mantine', 'murkrow'].every(m => species.id !== m) &&
```
Comment (MathyFurret) [inline data/mods/gen6/random-teams.ts:85]:
```suggestion
				!counter.get('Flying') && !['aerodactylmega', 'charizardmegay', 'mantine', 'murkrow'].includes(species.id) &&
```

**Review by MathyFurret — APPROVED** [review]
no obvious problems

File: data/mods/gen6/random-teams.ts (hunk 1 above)
Comment (ACakeWearingAHat) [reply data/mods/gen6/random-teams.ts]: (this was done but i can't mark as resolved)

**Pull request merged**